tests/phpunit/tests/user/wp-user/getUserBy.php

```diff
@@ -0,0 +1,194 @@
+<?php
+
+/**
+ * @group  user
+ * @covers WP_User::get_data_by
+ */
+class Tests_User_WpUser_GetDataBy extends WP_UnitTestCase {
+	protected static $user_id;
+	protected static $user_data = array(
+		'user_login'    => 'user1',
+		'user_nicename' => 'userone',
+		'user_pass'     => 'password',
+		'first_name'    => 'John',
+		'last_name'     => 'Doe',
+		'display_name'  => 'John Doe',
+		'user_email'    => 'jonny@battlefield3.com',
+		'user_url'      => 'http://tacos.com',
+		'role'          => 'contributor',
+		'nickname'      => 'Johnny',
+		'description'   => 'I am a WordPress user that cares about privacy.',
+	);
+
+	public static function wpSetUpBeforeClass( WP_UnitTest_Factory $factory ) {
+		self::$user_id = $factory->user->create( self::$user_data );
+	}
+
+	/**
+	 * @dataProvider data_id_field_for_valid_userid
+	 */
+	public function test_id_field_for_valid_userid( $cast_func ) {
+		$user_id = $cast_func( self::$user_id );
+
+		$user = WP_User::get_data_by( 'id', $user_id );
+		$this->assertInstanceOf( 'stdClass', $user, 'User is not an instance of stdClass' );
+		$this->assertSame( self::$user_id, (int) $user->ID, 'User ID does not match the user generated by the factory' );
+	}
+
+	/**
+	 * Data provider.
+	 *
+	 * @return array
+	 */
+	public function data_id_field_for_valid_userid() {
+		return array(
+			'integer user ID' => array( 'intval' ),
+			'string user ID'  => array( 'strval' ),
+			'float user ID'   => array( 'floatval' ),
+		);
+	}
+
+	/**
+	 * @dataProvider data_id_field_for_invalid_userid
+	 * @group        me
+	 */
+	public function test_id_field_for_invalid_userid( $user_id ) {
```

Comment:
```suggestion
	 *
	 * @param mixed $user_id An invalid user ID.
	 */
	public function test_id_field_for_invalid_userid( $user_id ) {
```
Parameters from data providers should be documented.